src/ngx_lua_resty_lmdb_module.c

```diff
@@ -289,26 +282,108 @@ static ngx_int_t ngx_lua_resty_lmdb_init_worker(ngx_cycle_t *cycle)
         block_size = EVP_CIPHER_block_size(cipher);
         if (block_size == 0) {
             ngx_log_error(NGX_LOG_CRIT, cycle->log, 0,
-                "unable to set LMDB encryption key, block size");
+                          "unable to set LMDB encryption key, block size");
             return NGX_ERROR;
         }
 
         rc = mdb_env_set_encrypt(lcf->env, lmdb_encrypt_func, &enckey, block_size);
         if (rc != 0) {
             ngx_log_error(NGX_LOG_CRIT, cycle->log, 0,
-                "unable to set LMDB encryption key: %s", mdb_strerror(rc));
+                          "unable to set LMDB encryption key: %s",
+                          mdb_strerror(rc));
             return NGX_ERROR;
         }
 
-        /* destroy data*/
-        ngx_explicit_memzero(lcf->key_data.data, lcf->key_data.len);
-        ngx_explicit_memzero(lcf->encryption_mode.data, lcf->encryption_mode.len);
+        /* worker will destroy secret data */
+        if (is_master == 0) {
+            ngx_explicit_memzero(lcf->key_data.data, lcf->key_data.len);
+            ngx_explicit_memzero(lcf->encryption_mode.data, lcf->encryption_mode.len);
 
-        ngx_str_null(&lcf->key_data);
-        ngx_str_null(&lcf->encryption_mode);
+            ngx_str_null(&lcf->key_data);
+            ngx_str_null(&lcf->encryption_mode);
+        }
+    }
+
+    return NGX_OK;
+}
+
+
+static ngx_int_t
+ngx_lua_resty_lmdb_remove_files(ngx_cycle_t *cycle, ngx_path_t *path)
+{
+    u_char            name_buf[NGX_MAX_PATH];
+    ngx_str_t        *name;
+    ngx_str_t         names[] = { ngx_string("/data.mdb"),
+                                  ngx_string("/lock.mdb"),
+                                  ngx_null_string,
+                                };
+
+    ngx_log_debug1(NGX_LOG_DEBUG_CORE, cycle->log, 0,
+                   "lmdb file remove: \"%V\"", &path->name);
+
+    /* try to remove a file or empty directory */
+    remove((const char *) path->name.data);
+
+    /* ensure lmdb directory exists */
+    mkdir((const char *) path->name.data, LMDB_DIR_MODE);
+
+    /* try to remove all lmdb files */
+    for (name = names; name->len; name++) {
+        ngx_snprintf(name_buf, NGX_MAX_PATH,
+                     "%V%V%Z", &path->name, name);
+
+        ngx_log_debug1(NGX_LOG_DEBUG_CORE, cycle->log, 0,
+                       "lmdb file remove: \"%s\"", name_buf);
+
+        remove((const char *) name_buf);
+    }
+
+    return NGX_OK;
+}
+
+
+static ngx_int_t
+ngx_lua_resty_lmdb_open_file(ngx_cycle_t *cycle,
+                             ngx_lua_resty_lmdb_conf_t *lcf,
+                             ngx_flag_t is_master)
+{
+    int                        rc;
+    int                        dead;
+    int                        tried = 0;
+
+    for (;;) {
```

Comment:
Why is this loop needed?

Reply:
The first time we try to create and open lmdb file, if it fails we will remove the files and try again.

Reply:
Remove the loop, using `goto retry`.